### PR TITLE
🪟 🧪 E2E Tests for auto-detect schema changes

### LINKED ISSUE #17936: Add E2E tests for auto-detect schema changes frontend
Blocked by:
- [x] #19878

### PR DESCRIPTION
**Pull request by edmundito**
## What
Closes #17936 

Adds E2E tests for auto-detect schema changes:

* Verify the status on the connection list page
* Verify that the user can save through the breaking and non-breaking flow
* Verify that the user can update the non-breaking preference

To make this possible, there were a few additions to the cypress framework:

* Adds the ability to create sources, destinations, and connections without using the UI
* Update the db query helpers to be able to generate SQL commands more easily

### CODE REVIEW

File: airbyte-webapp-e2e-tests/cypress/commands/utils/promise.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Converts Cypress chain to regular Promise.
+ */
+export const toPromise = <T>(chain: Cypress.Chainable<T>): Promise<T> => {
+  return new Cypress.Promise((resolve, reject) => {
```
Comment (krishnaglick) [inline airbyte-webapp-e2e-tests/cypress/commands/utils/promise.ts:5]:
```suggestion
  return new Cypress.Promise<T>((resolve, reject) => {
```

File: airbyte-webapp-e2e-tests/cypress/pages/replicationPage.ts [hunk 2]
```diff
@@ -118,14 +124,56 @@ export const searchStream = (value: string) => {
...
+};
+
+export const checkSchemaChangesDetectedCleared = () => {
+  cy.get(schemaChangesDetectedBanner).should("not.exist");
+  cy.get(schemaChangesBackdrop).should("not.exist");
+};
+
+export const clickSchemaChangesReviewButton = () => {
+  cy.get(schemaChangesReviewButton).click();
+  // cy.get(schemaChangesReviewButton).should("be.disabled");
```
Comment (krishnaglick) [inline airbyte-webapp-e2e-tests/cypress/pages/replicationPage.ts:173]: Why commented?
We need to enable the new table?

File: airbyte-webapp-e2e-tests/cypress/commands/db/queries.ts [hunk 3]
```diff
@@ -1,33 +1,68 @@
...
+
+export const insertIntoTable = (tableName: string, valuesByColumn: Record<string, unknown>): string => {
+  const keys = Object.keys(valuesByColumn);
+  const values = keys
+    .map((key) => valuesByColumn[key])
+    .map((value) => (typeof value === "string" ? `'${value}'` : value));
+
+  return `INSERT INTO ${tableName}(${keys.join(", ")}) VALUES(${values.join(", ")});`;
+};
+
```
Comment (dizel852) [inline airbyte-webapp-e2e-tests/cypress/commands/db/queries.ts:22]: 👍

**Review by dizel852 — COMMENTED** [review]
🔥 Happy to see so many good improvements - create a connection via API, instead of running the full flow of creation a new connection 👍 

Also tried to run new tests locally, but wasn't successful. Fails on `statusCell`:
image

File: airbyte-webapp-e2e-tests/cypress/pages/replicationPage.ts (hunk 2 above)
Comment (edmundito) [reply airbyte-webapp-e2e-tests/cypress/pages/replicationPage.ts:173]: This was related to not disabling the button in the loading state, which is fixed in #20737

**Comment by dizel852** [comment]
Tested locally, 1 test fails:
image

In the new table we use `<SyncModeSelect />`dropdown instead of the old one - `<SyncSettingsDropdown />`

**Comment by edmundito** [comment]
@dizel852 this is currently not compatible with the new table.

**Review by dizel852 — APPROVED** [review]
Retested locally with old stream table - works like a charm 👍 
Can't wait to start using the new test API 🔥

**Pull request merged**